coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
 import aiohttp
 import asyncio
 import json
-import traceback
-import urllib
-import urllib.request
 import logging
-import zstandard
-import lzma
 import csv
 import os
 import zipfile
-from collections import defaultdict
-
-TARGETS = [
-    ("logic/buildings.csv", "buildings.csv"),
-    ("logic/characters.csv", "characters.csv"),
-    ("logic/heroes.csv", "heroes.csv"),
-    ("logic/pets.csv", "pets.csv"),
-    ("logic/spells.csv", "spells.csv"),
-    ("logic/super_licences.csv", "supers.csv"),
-    ("logic/townhall_levels.csv", "townhall_levels.csv"),
-    ("logic/character_items.csv", "equipment.csv"),
-    ("localization/texts.csv", "texts_EN.csv"),
-]
-
-APK_URL = "https://d.apkpure.net/b/APK/com.supercell.clashofclans?version=latest"
-
-def get_fingerprint():
-    async def download():
-        async with aiohttp.request('GET', APK_URL) as fp:
-            c = await fp.read()
+import zstandard
+import lzma
+from pathlib import Path
+
+class StaticUpdater:
+    def __init__(self):
+
+        self.TARGETS = []
+        self.supported_languages: list[str] = []
+        self.USED_TIDS = set()
+
+        # keep the raw CSV files
+        self.KEEP_CSV = False
+        # keep the raw JSON files
+        self.KEEP_JSON = False
+        # removes any TIDs not used in the static files
+        self.PRUNE_TRANSLATIONS = True
+        # base path for the static files to be stored in
+        self.BASE_PATH = ""
+
+        self.FINGERPRINT = "475cb6a2d13043762034ddd6a198bad23e0782eb"
+        self.CLASH_VERSION = "" or "latest"
+        self.VERSION_PARAM = "version" if self.CLASH_VERSION == "latest" else "versionCode"
```

Comment:
this should be a property as changing the CLASH_VERSION after init will mess up the value.

We only need this in the download function or? so maybe just allow a version input there and determine the VERSION_PARAM there?